Modules/socketmodule.c

```diff
@@ -2116,12 +2120,18 @@ getsockaddrarg(PySocketSockObject *s, PyObject *args,
             }
             const char *straddr = PyBytes_AS_STRING(args);
             size_t len = PyBytes_GET_SIZE(args);
-            if (len >= sizeof(_BT_HCI_MEMB(addr, node))) {
+            if (strlen(straddr) != len) {
+                PyErr_Format(PyExc_OSError, "%s: "
```

Comment:
Some code raises OSError for invalid address, other raises more traditional TypeError and ValueError.

I think that it is better to raise TypeError (for invalid type) and ValueError (for too long bytes or embedded null byte). They are programming or user data errors, not produced by the OS API call. In future we can change other OSError to more suitable exception types.

Reply:
I changed the exception type to ValueError for "node too long" and "embedded null character".